### PR TITLE
Fixed bugs reported by scan-build from wazuh 3.12.3

### PR DESCRIPTION
**Pull request by danimegar**
|Related issue|
|---|
|#4990|

## Description
The bugs reported by scan-build have been fixed for Wazuh server, Mac agent and winagent. 

## Tests

- [x] Compilation on Linux
- [x] Compilation for Windows
- [x] Compilation on MAC OS X
- [x] Review logs syntax and correct language
- [x] Scan-build report for Linux
- [x] Scan-build report for Windows
- [x] Scan-build report for macOS
- [x] Coverity
- [x] Ruleset test

### CODE REVIEW

**Comment by danimegar** [comment]
## Reports:
Scan-build for server
server_scanbuild

Scan-build for winagent
winagent_scanbuild

Scan-build for mac agent

macagent_scanbuild

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 7d342962122de1ba33844e297f6dd0387c9d0f4e
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1769/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1769/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1769/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1769.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1769/Analysis_dynamic_valgrind-v3.13.0-B1769.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 7d342962122de1ba33844e297f6dd0387c9d0f4e
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1770/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1770/ossec.log)
**Coverage:** 

```json
Lines executed:27.00% of 3159

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1770/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1770.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1770/Analysis_dynamic_valgrind-v3.13.0-B1770.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 7d342962122de1ba33844e297f6dd0387c9d0f4e
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1775/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1775/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1775/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1775.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1775/Analysis_dynamic_valgrind-v3.13.0-B1775.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 7d342962122de1ba33844e297f6dd0387c9d0f4e
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1776/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1776/ossec.log)
**Coverage:** 

```json
Lines executed:27.00% of 3159

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1776/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1776.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1776/Analysis_dynamic_valgrind-v3.13.0-B1776.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 44fa0e63f18c8408c2e0db398c73a844f430e938
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1791/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1791/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1791/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1791.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1791/Analysis_dynamic_valgrind-v3.13.0-B1791.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 44fa0e63f18c8408c2e0db398c73a844f430e938
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1792/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1792/ossec.log)
**Coverage:** 

```json
Lines executed:26.97% of 3159

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1792/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1792.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1792/Analysis_dynamic_valgrind-v3.13.0-B1792.log)

File: src/analysisd/rules.c [hunk 1]
```diff
@@ -1404,11 +1404,12 @@ int Rules_OP_ReadRules(const char *rulefile)
                                 merror("Invalid option '%s' for "
                                 "rule '%d'", mitre_opt[ind]->element,
                                 config_ruleinfo->sigid);
-
-                                for (l = 0; config_ruleinfo->mitre_id[l] != NULL; l++) {
-                                    os_free(config_ruleinfo->mitre_id[l]);
+                                if (config_ruleinfo->mitre_id) {
```
Comment (vikman90) [inline src/analysisd/rules.c:1407]: We could replace this block with:
```c
free_strarray(config_ruleinfo->mitre_id);
```

In fact, this function already checks that `mitre_id` is not null;

File: src/logcollector/read_win_event_channel.c [hunk 2]
```diff
@@ -310,7 +310,7 @@ int update_bookmark(EVT_HANDLE evt, os_channel *channel)
         goto cleanup;
     }
 
-    if ((buffer = calloc(size, sizeof(char))) == NULL) {
+    if (buffer = calloc(size, sizeof(wchar_t)), buffer == NULL) {
```
Comment (vikman90) [inline src/logcollector/read_win_event_channel.c:313]: According to the documentation of [EvtRender](https://docs.microsoft.com/en-us/windows/win32/api/winevt/nf-winevt-evtrender), the size of the buffer is measured in bytes.

**Comment by danimegar** [comment]
## Coverity

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** bd585ee4fddbeb92f9ca6544bd1f3dea703be5df
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1853/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1853/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1853/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1853.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1853/Analysis_dynamic_valgrind-v3.13.0-B1853.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** bd585ee4fddbeb92f9ca6544bd1f3dea703be5df
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1854/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1854/ossec.log)
**Coverage:** 

```json
Lines executed:27.00% of 3159

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1854/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1854.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1854/Analysis_dynamic_valgrind-v3.13.0-B1854.log)

**Pull request merged**